go/kbfs/libkbfs/prefetcher.go

```diff
@@ -1001,8 +1032,13 @@ func (p *blockPrefetcher) ProcessBlockForPrefetch(ctx context.Context,
 	ptr BlockPointer, block Block, kmd KeyMetadata, priority int,
 	lifetime BlockCacheLifetime, prefetchStatus PrefetchStatus,
 	action BlockRequestAction) {
+	obseleted := make(chan struct{})
 	req := &prefetchRequest{ptr, block.NewEmptier(), kmd, priority, lifetime,
-		prefetchStatus, action, nil}
+		prefetchStatus, action, nil, obseleted}
+	p.queuedPrefetchHandlesLock.Lock()
+	p.queuedPrefetchHandles[req.ptr.ID] = obseleted
```

Comment:
I think this would make more sense in `triggerPrefetch()` (which is called in one other place).  It can add the channel to the request there, rather than during construction above.